frame/collective/src/lib.rs

```diff
@@ -659,17 +681,21 @@ decl_module! {
 				Self::deposit_event(RawEvent::Closed(proposal_hash, yes_votes, no_votes));
 				let (proposal_weight, proposal_count) =
 					Self::do_approve_proposal(seats, voting, proposal_hash, proposal);
-				return Ok(Some(
-					T::WeightInfo::close_approved(len as u32, seats, proposal_count)
-						.saturating_add(proposal_weight)
-				).into());
+
+				T::WeightInfo::close_approved(len as u32, seats, proposal_count)
+					.saturating_add(proposal_weight)
 			} else {
 				Self::deposit_event(RawEvent::Closed(proposal_hash, yes_votes, no_votes));
 				let proposal_count = Self::do_disapprove_proposal(proposal_hash);
-				return Ok(Some(
-					T::WeightInfo::close_disapproved(seats, proposal_count)
-				).into());
+
+				T::WeightInfo::close_disapproved(seats, proposal_count)
+			};
+
+			for member in absentation_members {
+				T::OnAbsentation::on_absentation(&member);
```

Comment:
The cost of on_absentation is not always part of the benchmark. for example the benchmark for close_disapproved have always less than 4 absentation.

Not sure how to this fix it though, maybe add some comment in the doc. or introduce some new benchmark.

Reply:
I think right now the benchmark is assuming `T::OnAbsentation` is `()`. For chains setting this to custom values, they probably need to provide their own `WeightInfo`.

Reply:
benchmarks are run for the runtime, if user run benchmarks with a runtime where OnAbsentation is not no-op then the benchmark will run the logic of OnAbsentation. 

I think current strategy is to have user running benchmarks for their runtime, and just taking the resulting WeightInfo. Thus benchmarks should be written to cover all potential cost as far as possible.

Reply:
though if benchmarking it is not easy I'm ok to just a comment somewhere saying that the benchmark doesn't handle the cost of OnAbsentation and that any costy OnAbsentation implementation must be careful and might need to assign weight manually.

Reply:
> I'm ok to just a comment somewhere saying that the benchmark doesn't handle the cost of OnAbsentation and that any costy OnAbsentation implementation must be careful and might need to assign weight manually.

Sorry I missed your review on this. I'll add the comment.

Reply:
ah I could have ping again also, all good now :-)